src/main/scala/com/github/swagger/akka/SwaggerHttpService.scala

```diff
@@ -99,6 +98,12 @@ trait SwaggerHttpService extends Directives {
     }
   }
 
+  private def filteredSwagger: Swagger = {
+    val swagger: Swagger = reader.read(toJavaTypeSet(apiTypes).asJava)
+    swagger.setDefinitions(swagger.getDefinitions.asScala.filterNot { case (modelName, _) => unwantedDefinitions.contains(modelName) }.asJava)
```

Comment:
.getDefinitions.getKeySet would probably be simpler - could you also use `definitionName` instead of `modelName`?